code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -266,24 +266,29 @@
 						visible_message(SPAN_DANGER("[src]'s [W] goes off during the struggle!"))
 						return W.afterattack(target,src)
 
-			var/randn = rand(1, 100)
-			randn = max(1, randn - H.stats.getStat(STAT_ROB))
-
-			if(randn <= 50)
-				//See about breaking grips or pulls
-				if(break_all_grabs(M))
-					playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
-					return
-
-				//Actually disarm them
-				for(var/obj/item/I in holding)
-					if(I && src.unEquip(I))
-						visible_message(SPAN_DANGER("[M] has disarmed [src]!"))
+			//Actually disarm them
+			var/rob_attacker = (50 / (1 + 150 / (min(1, H.stats.getStat(STAT_ROB)))) + 20) //soft capped amount of recoil that attacker deals
+			var/rob_target = max(0, min(400, src.stats.getStat(STAT_ROB))) //hard capped amount of recoil the target negates upon disarming. 400 - no recoil
+			var/recoil_damage = (rob_attacker * (1 - (rob_target / 400))) //recoil itself
+			for(var/obj/item/I in holding)
+				src.external_recoil(recoil_damage)
+				if(src.recoil >= 60) //disarming
```

Comment:
No need for the `src` everywhere, also you should put this if check before the for loop.

Reply:
the if check needs to be there because of external_recoil(recoil_damage). If it's outside of for, you either will add recoil to people who aren't even holding anything, or disarming won't work at all because the recoil amount remains the same